### PR TITLE
Fix/17063/avoid undefined args

### PR DESCRIPTION
**Pull request by evont**
Issue: #17063

## What I did
fix args values not present in url are casted to !undefined for controls of type select and radio [issue](https://github.com/storybookjs/storybook/issues/17063) 

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Review by kylegach — COMMENTED** [review]
Thank for fixing this important issue, @evont! And with your very first contribution. 👏 

@shilman @tmeasday — This change seems sound to me, but the args code in general is likely quite nuanced and I think it would benefit from you taking a look as well. Also, is it OK that this PR is against `main` instead of `next`?

**Comment by kylegach** [comment]
@evont — I'd do this myself, but I cannot push to your fork's branch. Can you please add this story, to the `examples/official-storybook/stories/core/args.stories.js` file?

```js
export const TestUndefinedArgs = Template.bind({});
TestUndefinedArgs.args = {
  first: 'Bob',
  last: 'Miller',
  foo: 'bar',
};
TestUndefinedArgs.argTypes = {
  first: {
    options: ['Bob', 'Alice'],
  },
  last: {
    options: ['Miller', 'Meyer'],
  },
  foo: {
    control: { type: 'text' },
  },
};
```

That will allow maintainers to easily test your change (use the same repro steps as the linked issue) in the deployed Storybook. Thanks!

**Review by shilman — CHANGES_REQUESTED** [review]
Hi @evont @kylegach -- thanks for the fix! Indeed all PRs should go against `next` not `main` unless it's a special case, which AFAICT this is not. Please rebase & retarget the PR. 🙏

File: lib/store/src/args.ts [hunk 1]
```diff
@@ -101,7 +101,8 @@ export const validateOptions = (args: Args, argTypes: ArgTypes): Args => {
     const isValidArray = isArray && invalidIndex === -1;
 
     if (args[key] === undefined || options.includes(args[key]) || isValidArray) {
```
Comment (tmeasday) [inline lib/store/src/args.ts:103]: This check for `args[key] === undefined` seems a little pointless if we later do `if (key in args)`. Do we understand why the code was there in the first place? We should get rid of it, and perhaps figure that out.

Comment (tmeasday) [reply lib/store/src/args.ts:103]: Or does it remain here to avoid the warning lower down?

**Review by tmeasday — CHANGES_REQUESTED** [review]
Thanks for the contribution @evont. Can you please add test case(s) that demonstrate the original problem?

**Comment by evont** [comment]
> @evont — I'd do this myself, but I cannot push to your fork's branch. Can you please add this story, to the `examples/official-storybook/stories/core/args.stories.js` file?
> 
> ```js
> export const TestUndefinedArgs = Template.bind({});
> TestUndefinedArgs.args = {
>   first: 'Bob',
>   last: 'Miller',
>   foo: 'bar',
> };
> TestUndefinedArgs.argTypes = {
>   first: {
>     options: ['Bob', 'Alice'],
>   },
>   last: {
>     options: ['Miller', 'Meyer'],
>   },
>   foo: {
>     control: { type: 'text' },
>   },
> };
> ```
> 
> That will allow maintainers to easily test your change (use the same repro steps as the linked issue) in the [deployed Storybook](https://deploy-preview-17973--storybookjs.netlify.app/official-storybook/?path=/story/core-args--test-undefined-args). Thanks!

Sure, with pleasure, I have added it and push it to the branch, except I added `control: { type: 'select' }` because it is odd that if I don't do that, the `options` will not transform into `select` but `text`

**Comment by evont** [comment]
Sorry for my mistake, fork from main branch and rebase to next will be that tough, so I decided to fork form next and make a new PR, please check in this [new PR](https://github.com/storybookjs/storybook/pull/18135)
@tmeasday @kylegach @shilman

**Comment by shilman** [comment]
Thanks @evont! Closing this in favor of #18135

**Pull request closed without merging**